Classes/GarbageCollector.php

```diff
@@ -215,6 +219,33 @@ protected function deleteIndexDocuments($table, $uid)
     }
 
     /**
+     * Delete all index documents that are related to a TYPO3 "page"
+     *
+     * @param $uid
+     */
+    protected function deleteRecordsOfPage($uid)
+    {
+        /** @var \ApacheSolrForTypo3\Solr\ConnectionManager $connectionManager */
+        $connectionManager = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\ConnectionManager');
+        $indexQueue = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\IndexQueue\\Queue');
+        $site = Site::getSiteByPageId($uid);
+
+        // a site can have multiple connections (cores / languages)
+        $solrConnections = $connectionManager->getConnectionsBySite($site);
+
+        /** @var  $solr */
+        foreach ($solrConnections as $solr) {
+            $response = $solr->search('pid:' . intval($uid), 0, 10000);
```

Comment:
Yeah the problem is that the page that holds the entries must not be in the index queue (page of type folder) and as already mentioned there is no easy way to find out which records belong to a specific page.

So i used this approuch to clean the index form records that should not be there anymore.

I don't see good way to get all related recoreds for now... What would you suggest?